lib/std/crypto/salsa20.zig

```diff
@@ -0,0 +1,430 @@
+const std = @import("std");
+const crypto = std.crypto;
+const debug = std.debug;
+const math = std.math;
+const mem = std.mem;
+
+const Poly1305 = crypto.onetimeauth.Poly1305;
+const Blake2b = crypto.hash.blake2.Blake2b;
+const X25519 = crypto.dh.X25519;
+
+const Salsa20NonVecImpl = struct {
+    const BlockVec = [16]u32;
+
+    fn initContext(key: [8]u32, d: [4]u32) BlockVec {
+        const c = "expand 32-byte k";
+        const constant_le = comptime [4]u32{
+            mem.readIntLittle(u32, c[0..4]),
+            mem.readIntLittle(u32, c[4..8]),
+            mem.readIntLittle(u32, c[8..12]),
+            mem.readIntLittle(u32, c[12..16]),
+        };
+        return BlockVec{
+            constant_le[0], key[0],         key[1],         key[2],
+            key[3],         constant_le[1], d[0],           d[1],
+            d[2],           d[3],           constant_le[2], key[4],
+            key[5],         key[6],         key[7],         constant_le[3],
+        };
+    }
+
+    const QuarterRound = struct {
+        a: usize,
+        b: usize,
+        c: usize,
+        d: u6,
+    };
+
+    inline fn Rp(comptime a: usize, comptime b: usize, comptime c: usize, comptime d: u6) QuarterRound {
```

Comment:
why is this inline? why are the parameters comptime?
Why does this even exist at all? just to make the following function more nicely formatted?

Reply:
Yes, just to make the key schedule more readable, and the code look similar to the ChaCha20 one.
Given the purpose of the function, callout it with a non-comptime parameter would be unintentional. But well, `comptime` can be removed no matter what.